### PR TITLE
[2.13.x] DDF-4133 Solr auth changes

### PR DESCRIPTION
**Pull request by ahoffer**
##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Porting changes from master branch to release (2.13.x) branch.
Link to 3 master branch PRs:
- https://github.com/codice/ddf/pull/4083
- https://github.com/codice/ddf/pull/4142
- https://github.com/codice/ddf/pull/4160

The master branch PRs are based on changes from PR that breaks up the solr-dependencies JAR: https://github.com/codice/ddf/pull/4134. That PR affects many, many additional files and was not backported. Therefore, there are some minor differences between the master and release branch implementations.

#### What does this PR do?
- Changes the DDF Solr connection from two-way TLS to one-way TLS
- Adds basic authentication for the DDF to prove its identity to Solr
- Adds a module to generate a basic auth password, change the password in Solr, encrypt the password, and save the encrypted password to a file.
- Introduces new system properties to control behavior (see testing section)

#### Who is reviewing it? 
@austinsteffes 
@tyler30clemens 
#### Select relevant component teams: 
@codice/security 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.
@brjeter
@vinamartin

#### How should this be tested?
1. Review documentation changes in section 7.5.1.2.2. Solr Server Password Management of the documentation. Perform the configuration task described in the section. Confirm that the system behaves as documented.
1. Change the values or combination of values of the new system properties and determine if they work as described in the documentation section Managed Solr Server Properties
solr.useBasicAuth
solr.username
solr.password
solr.attemptAutoPasswordChange

#### Checklist:
- [X] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by ahoffer** [comment]
build now

File: distribution/docs/src/main/resources/content/_configuring/hardening-solr.adoc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ {external-link}. Here is an example that uses the
+command line utility `curl` to change the password from `autogenerated` to `newpassword`:
+
+ curl -k -u "ddf:autogenerated" "${secure_url}/solr/admin/authentication" -H 'Content-type:application/json' -d "{ 'set-user': {'ddf' : 'newpassword'}}"
++
+. Encrypt the password using the <<{integrating-prefix}encryption_service,Encryption Service>>.
+The encryption command enciphers the password. It is safe to save the peristed password in a file.
+. Update property `solr.password` in the file ${home_directory}/etc/custom.system.properties` to
+be the ouput from the encryption command. Be sure to include `ENC(` and `)` characters produced by
+the encryption comand. Note that the default password is not enclosed in `ENC()` because that
```
Comment (tyler30clemens) [inline distribution/docs/src/main/resources/content/_configuring/hardening-solr.adoc]: command typo

File: distribution/solr-distro/assembly.xml [hunk 2]
```diff
@@ -43,6 +47,11 @@
     </dependencySets>
     <fileSets>
         <fileSet>
+            <!-- The order of the <fileSet> elements is important. The assumbly plugin will not
```
Comment (tyler30clemens) [inline distribution/solr-distro/assembly.xml]: Is assumbly a typo for assembly?

File: distribution/docs/src/main/resources/content/_configuring/hardening-solr.adoc [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
++
+. Edit the file `${home_directory}/etc/custom.system.properties` and change the value of
+`solr.password` to `autogenerated`.
+. Optional: <<_do_not_autogenerate_a_solr_password, Prevent ${branding} from automatically changing the Solr password>>.
+
+==== Removing Basic Authentication from Solr
+
+To disable Solr's basic authentication mechanism, rename or remove the file
+`${home_directory}/solr/server/solr/security.json` and restart Solr. The file `security.json`
+configures Solr to use basic authetnication and defines Solr users. If the file is not present,
```
Comment (tyler30clemens) [inline distribution/docs/src/main/resources/content/_configuring/hardening-solr.adoc]: authentication typo

File: distribution/solr-distro/assembly.xml [hunk 4]
```diff
@@ -59,6 +68,15 @@
             </includes>
             <fileMode>0755</fileMode>
```
[the remark this answers is not in the document]
Comment (ahoffer) [reply distribution/solr-distro/assembly.xml]: I have some regret about changing master to 0750. It was 0755 and I decided, for now particular reason, to make it a tiny bit more restrictive. I'm not sure I should have done that.

File: features/security/src/main/feature/feature.xml [hunk 5]
```diff
@@ -71,6 +71,7 @@
     <feature name="security-encryption" version="${project.version}"
              description="Security Encryption">
         <feature>wss4j</feature>
+        <feature>apache-commons</feature>
```
[the remark this answers is not in the document]
Comment (ahoffer) [reply features/security/src/main/feature/feature.xml]: On the master branch, I did not see that there was a feature. But in the interest of consistency and not loading unnecessary code, I'll use the bundle.

**Comment by ahoffer** [comment]
build now

File: distribution/solr-distro/assembly.xml [hunk 6]
```diff
@@ -59,6 +68,15 @@
             </includes>
```
[the remark this answers is not in the document]
Comment (ahoffer) [reply distribution/solr-distro/assembly.xml]: I decided to leave the instructions specific to individual files. The comments of why we replace certain files from the upstream Solr distro seemed useful.

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

**Comment by ahoffer** [comment]
build now

File: distribution/docs/src/main/resources/content/_running/solr-service.adoc [hunk 7]
```diff
@@ -53,11 +53,17 @@ These instructions are for unix operating systems running the systemd
 
 . If ${branding} is running, stop it.
 . Edit `${home_directory}/etc/custom.system.properties` and set `start.solr=false`.
+. Edit the file `${home_directory}/solr/services/solr.service`
+  .. Edit the property `Environment=JAVA_HOME` and replace `<JAVA_HOME>` with the absolute path
+  to the directory where the Java Runtime Environment is installed.
+  .. Edit the property `ExecStart` and replace ${home_directory} with the absolute path to the `ddfsolr` file.
+  .. Edit the property `ExecStopt` and replace ${home_directory} with the absolute path to the `ddfsolr` file.
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/content/_running/solr-service.adoc]: ❓ ExecStopt? or ExecStop?

**Comment by tyler30clemens** [comment]
Windows test: verified was able to upload to the catalog and search the catalog with no issues

**Comment by vinamartin** [comment]
build now

**Comment by garrettfreibott** [comment]
build now

**Pull request merged**